[Migrated to PR #676] Decouples Mouse Crosshair Speed [Aiming] from Mouse Speed [Camera]

AL2009man opened the pull request:
In the past, Mouse's Crosshair Speed is treated as a Multiplier, however, it's not communicated to the end-user, ultimately believing the user may want to increase the mouse sensitivity to...10.00 while keeping the Crosshair Sens around 3.00. The user found out that it increased the Crosshair Speed as a result

This addresses https://github.com/fgsfdsfgs/perfect_dark/issues/595, which the Crosshair Speed becomes independent of the general Mouse Camera system, allowing Joanna Dark to mix-and-match sensitivity styles.

As this PR will once again override user's muscle memory, this will require playtesting from KB/M users on how the overall Crosshair Scaling feels. the end goal is getting a sweet spot between regular mouse camera and crosshair camera within a default slider of 2.50, this will be tricky to nail. this'd be a Draft PR for the time being.

~~edit: this also includes a unified Camera Sensitivity scaling system, makes thing easier to manage while potentially introducting future input-related features like Flick Stick~~

AL2009man committed:
initial attempt at decoupling mouse crosshair aim

AL2009man committed:
Decoupled Mouse Crosshair Speed entirely

AL2009man committed:
removed mouse sens cap code leftover

AL2009man committed:
cleaned up Mouse Crosshair code

AL2009man committed:
changed default Mouse Crosshair Speed to 3.50, changed scaling method

This will now default to 3.50, but overall tweaking is still unfinished and will be the case until further notice.

AL2009man committed:
Merge branch 'port' into MouseCrosshairFix

AL2009man committed:
Introduced unified Camera Sensitivity Scale, changed overall Crosshair Mouse Sens

backported the unified Camera sens scale system from a separate PR over to here, while changing overall Crosshair Mouse scaling to closely mirror default mouse camera sens.

AL2009man committed:
slightly revised notes on scaling factor

AL2009man commented:
took some time adjusting of the Crosshair Mouse Speed scaling. 

it should now be closer to the main Camera Mouse movement (at default 2.5 settings) while ensuring the muscle memory portion is consistent for those who are already gotten used to it. Thus: the final mouse crosshair speed default will be 2.50.

Additionally, this PR will include a Unified Camera scaling system across all Input styles, allowing me to implement future Input features like Flick Stick.

AL2009man committed:
Reverting Unified camera scaling for now

found out a separate issue while revising the camera scaling, for now: this will be reverted back to how it was.

AL2009man committed:
restored leftover from older fix

AL2009man committed:
fix regression on MouseSpeedX/Y slider

AL2009man committed:
Merge branch 'fgsfdsfgs:port' into MouseCrosshairFix

AL2009man committed:
Merge branch 'fgsfdsfgs:port' into MouseCrosshairFix

AL2009man committed:
whitespace fix yet again

Visual Studios 2022 Communty loves whitespace!

AL2009man committed:
fixed `inputMouseGetAbsScaledDelta` regression during the git push/scan process

AL2009man commented:
> Please fix up the whitespace in all of your changes. Seems like you have it set to use two tabs instead of one for indents or something.

had to do last-minute whitespacing fix ~~(i'm switching to Visual Studios Code)~~, while doing regression fixes.

fgsfdsfgs commented:
> but I did found a bug. setting Mouse Camera to 0.00 causes Mouse Crosshair to not work. remnants of the old system. so i'll need to go ahead and get _that_ fixed soonish (got IRL Stuffs to do, but I'll come back a bit later.)

Well, if you set the sensitivity to 0, that's kind of expected, isn't it. Unless I'm misunderstanding.

Can you also nuke `mouseaimspeedx` and `mouseaimspeedy` from the extcfg struct since they're not used anymore?

AL2009man committed:
replaced old mouse aim code with the new system

AL2009man commented:
> > but I did found a bug. setting Mouse Camera to 0.00 causes Mouse Crosshair to not work. remnants of the old system. so i'll need to go ahead and get _that_ fixed soonish (got IRL Stuffs to do, but I'll come back a bit later.)
> 
> Well, if you set the sensitivity to 0, that's kind of expected, isn't it. Unless I'm misunderstanding.
> 
> Can you also nuke `mouseaimspeedx` and `mouseaimspeedy` from the extcfg struct since they're not used anymore?

it should be replaced with the new version. it can't be exactly nuked, tho

fgsfdsfgs commented:
> it should be replaced with the new version. it can't be exactly nuked, tho

Why? There's no more purpose in having an additional aim multiplier in the Game settings if you have the actual sensitivity in Input settings. And it looks like you've removed its usage from the aim code.

AL2009man committed:
Crosshair Sensitivity is no longer tied to framerate

AL2009man commented:
> > it should be replaced with the new version. it can't be exactly nuked, tho
> 
> Why? There's no more purpose in having an additional aim multiplier in the Game settings if you have the actual aim sensitivity in Input settings. And it looks like you've removed its usage from the aim code.

compiling issues, but i am confident that there shouldn't be any aim multipliers beyond Crosshair Aiming being tied to framerate (

AL2009man commented:
Note: do not merge until https://github.com/fgsfdsfgs/perfect_dark/pull/630, consider it a safety-net in case of conflictions while rapid-merging

AL2009man committed:
completely nuke "extcfg struct" on old variable

Inorder to be more consistent with the previous `mouseaimspeedx/y`,  the entire thing has been "nuked".

AL2009man committed:
fixed Mouse Crosshair Sens forcing to 0 if Mouse Speed is set to 0

This fixes a problem where Mouse Crosshair Sensitivity is forcefully set to 0 when Mouse Sensitivity is also set to 0. This should let you keep the same mouse crosshair sens but turn off Mouse Sens and might prefer using keyboard keys for camera movement?

AL2009man commented:
> > but I did found a bug. setting Mouse Camera to 0.00 causes Mouse Crosshair to not work. remnants of the old system. so i'll need to go ahead and get _that_ fixed soonish (got IRL Stuffs to do, but I'll come back a bit later.)
> 
> Well, if you set the sensitivity to 0, that's kind of expected, isn't it. Unless I'm misunderstanding.

turns out, it was that was causing this particular issue.

```
		allowmcross = (PLAYER_EXTCFG().mouseaimmode == MOUSEAIM_CLASSIC) &&
			(movedata.freelookdx || movedata.freelookdy || g_Vars.currentplayer->swivelpos[0] || g_Vars.currentplayer->swivelpos[1]);
```
 
simply removing the one below, like this:

```
		inputMouseGetScaledDelta(&movedata.freelookdx, &movedata.freelookdy);
		allowmcross = (PLAYER_EXTCFG().mouseaimmode == MOUSEAIM_CLASSIC);

```
addresses the issue, but this might need some additional reviewing to see if something feels...off.

 
> Can you also nuke `mouseaimspeedx` and `mouseaimspeedy` from the extcfg struct since they're not used anymore?

I believe i got it solved, while also maintaining backwards compatibility with pd.ini. This might break everyone's muscle memory in the process.

regardless: don't merge this PR until https://github.com/fgsfdsfgs/perfect_dark/pull/631 get merged in first, consider it a safety-net in case of conflicts while rapid-merging.

AL2009man committed:
change Y and X to match original repository format

AL2009man committed:
Centralized Crosshair logic

overhauled the entire Crosshair logic, while adjusting deltacrosshair scaling to reflect the recent changes

AL2009man committed:
reverted framerate-tied centralized

AL2009man committed:
Restored movedata.freelook joystick code
